### PR TITLE
Extract TestCluster interface

### PR DESCRIPTION
**Pull request by camerski**
Formalizing the TestCluster interface now that we have multiple people working on TestCluster implementations.

### Testing
Manually verified that `test.py` still works.

### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: bundle-workflow/python/test_workflow/test_cluster.py [hunk 1]
```diff
@@ -3,8 +3,26 @@
 import urllib.request
 import shutil
 import subprocess
+import abc
 
-class LocalTestCluster:
+class TestCluster(abc.ABC):
```
Comment (owaiskazi19) [inline bundle-workflow/python/test_workflow/test_cluster.py:8]: Can we pull the interface in a separate file? The other clusters like PerfTest, IntegTest can use the same

Comment (camerski) [reply bundle-workflow/python/test_workflow/test_cluster.py:8]: Sure

**Review by owaiskazi19 — APPROVED** [review]

File: bundle-workflow/python/test_workflow/test_cluster.py [hunk 2]
```diff
@@ -1,52 +1,41 @@
...
-                self.process.wait(10)
-            except TimeoutExpired:
-                print('Process failed to terminate even after SIGKILL')
-                raise
-        finally:
-            print(f'Process terminated with exit code {self.process.returncode}')
-            self.stdout.close()
-            self.stderr.close()
-            self.process = None
+class ClusterCreationException(Exception):
```
Comment (owaiskazi19) [inline bundle-workflow/python/test_workflow/test_cluster.py:37]: nit: Can't we use this class to throw error while creating the cluster?

Comment (camerski) [reply bundle-workflow/python/test_workflow/test_cluster.py:37]: That's the idea, yeah. I'll update LocalTestCluster to throw this exception when it fails.

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**